sdk/storage/src/queue/clients/queue_name_client.rs

```diff
@@ -1,86 +1,70 @@
 use crate::core::Client;
 use crate::queue::clients::QueueServiceClient;
+use crate::queue::PopReceipt;
 use crate::requests;
-use crate::{HasStorageClient, IntoQueueNameClient, QueueNameService, WithQueueNameClient};
-use azure_core::No;
+use crate::HasStorageClient;
 use std::borrow::Cow;
 use std::fmt::Debug;
 
 #[derive(Debug, Clone)]
-pub struct QueueNameClient<'a, 'b, C>
+pub struct QueueNameClient<C>
```

Comment:
It's not a bad idea. I wonder if we should also add `Send + Sync` to avoid limiting ergonomics of these structs. I'll take a stab at this.

Reply:
In Cosmos we're wrapping the HttpClient like this: `  http_client: Arc<Box<dyn HttpClient>>,`

Reply:
I like this a lot, but I would like to have a separate PR because it entails changing all the storage account functions (not just the queue ones). It's also needed to remove the hard dependency from `Hyper`.